apis/nucleus/src/components/selections/SelectedFields.jsx

```diff
@@ -15,27 +15,33 @@ import More from './More';
 const MIN_WIDTH = 120;
 const MIN_WIDTH_MORE = 72;
 
-function collect(qSelectionObject, fields, state = '$') {
-  qSelectionObject.qSelections.forEach((selection) => {
-    const name = selection.qField;
-    const field = (fields[name] = fields[name] || { name, states: [], selections: [] }); // eslint-disable-line
-    if (field.states.indexOf(state) === -1) {
-      field.states.push(state);
-      field.selections.push(selection);
-    }
-  });
-}
-
 function getItems(layout) {
   if (!layout) {
     return [];
   }
   const fields = {};
+
+  function collectFields(qSelectionObject, state) {
+    qSelectionObject.qSelections.forEach((selection) => {
+      const name = selection.qField;
+      let field = fields[name];
+      if (field === undefined) {
+        const label =
+          selection.qDimensionReferences?.find((element) => element.qLabel)?.qLabel || selection.qReadableName || name;
+        field = { name, label, states: [], selections: [] };
+        fields[name] = field;
+      }
+      if (field.states.indexOf(state) === -1) {
+        field.states.push(state);
+        field.selections.push(selection);
+      }
```

Comment:
I think we can include this within line 31? (at least only `state` part of it)

Here is what i thought about it: 
1. `const fields = {};` is empty when we enter `qSelectionObject.qSelections.forEach(...)` loop. 
2.  then line 27: `let field = fields[name];` will always be `undefined` in the first iteration. 
3.  then we will fall into if check at line 28
4. within that we will associate states as [] and selections as [] (line 31)
5. after that if check, we will fall into indexOf check to see if that associated states dont have (given) state (line 34) which if we had field as undefined (which we do according to step 2) -> then it always going to be an empty array, right? meaning we will fall into if case at line 34 only at the first iteration
so we could safely say line 31 should be something like: 
```
field = { name, label, states: [state], selections: [selection] }
```

because otherwise in other iterations, even if we find the `fields[name]` again (due to adding `field` into `fields[name]` in line 32) it will going to already have `fields.state` and `fields.selection` since we already associated them in the first iteration when they were empty array! and as result we will never going to fall into if check at line 34 again.

Also thinking a bit more about it, why we do `field.selections.push(selection);` always depending on when we don't have state in `fields.states`? 
what I mean is, if we already have state in `field.states`, will other selection going to be skipped and not get added to the `field.selections` array?

The main thing got me interested about why we should have the if check at line 34 was that: 
if we have field as undefined (line 28) we will have states as an empty array, and if and only if we are in that case then line 34 will be true (`field.states.indexOf(state) === -1`) because otherwise, it's already associated in the first iteration!


---
*I could be wrong on this or miss some info about the entire logic, but just sharing my though about it

Reply:
I gave all of this some thought, and more importantly: I read the code that sets up the structures of qSelectionObject.
There is one qSelectionObject for the default state $ and then an array with one qSelectionObject for each alternate state. There should be no duplicates.
I therefore simplified the code, adding each state and selection to the field.states and field.selections arrays. (Since there are no duplicate states.)